### ISSUE TITLE
Should there be a helper function for print methods

### ISSUE THREAD

**Issue by EmilHvitfeldt** [tidymodels/parsnip#567]
All the print methods follow this pattern

https://github.com/tidymodels/parsnip/blob/af46f571c4e7ea07d7a707bc1a12e3567ff93e5e/R/boost_tree.R#L86-L95

What about we create a wrapper so we can do

```r
print_helper <- function(name, x, ...) {
  cat(name, " Specification (", x$mode, ")\n\n", sep = "")
  model_printer(x, ...)
  
  if (!is.null(x$method$fit$args)) {
    cat("Model fit template:\n")
    print(show_call(x))
  }
  invisible(x)
}

print.decision_tree <- function(x, ...) {
  print_helper("Decision Tree Model", x, ...)
}
```

**Comment by juliasilge** [comment]
This would make it way easier to start using cli eventually, for example.

**Comment by EmilHvitfeldt** [comment]
The null model print is substantially different and should be put back in line with the rest

https://github.com/tidymodels/parsnip/blob/27d5b6b17a1a89ab058ed793a005b7eaee0a6d0e/R/nullmodel.R#L79-L95

And in its current state, the print method doesn't print that it is a null model

``` r
library(parsnip)
null_model()
#> Model Specification (classification)
```

**Comment by simonpcouch** [comment]
Ah, wasn't automatically closed with #739—doing so now. :)
